spring-integration-redis/src/test/java/org/springframework/integration/redis/outbound/ReactiveRedisStreamMessageHandlerTests.java

```diff
@@ -32,4 +83,142 @@ public void nullConnectionFactoryTest() {
 				.isThrownBy(() -> new ReactiveRedisStreamMessageHandler("stream", null));
 	}
 
+
+	@Test
+	@RedisAvailable
+	public void simpleStringInsertionTest() {
+		String streamKey = "myStream";
+		String messagePayload = "Bonjour à tous les confinés";
+
+		handlerAdapter.handleMessage(new GenericMessage<>(messagePayload));
+
+		RedisSerializationContext<String, String> serializationContext = redisStringOrJsonSerializationContext(true, null);
+
+		ReactiveRedisTemplate<String, String> template = new ReactiveRedisTemplate(redisConnectionFactory, serializationContext);
+
+		ObjectRecord<String, String> record = template.opsForStream().read(String.class, StreamOffset.fromStart(streamKey)).blockFirst();
+		assertThat(record.getStream()).isEqualTo(streamKey);
+		assertThat(record.getValue()).isEqualTo(messagePayload);
+		template.delete(streamKey).block();
+	}
+
+	@Test
+	@RedisAvailable
+	public void integrationStreamOutboundTest() {
+		String streamKey = "myStream";
+		String messagePayload = "Bonjour à tous les confinés";
+
+		messageChannel.send(new GenericMessage<>(messagePayload));
+
+		RedisSerializationContext<String, String> serializationContext = redisStringOrJsonSerializationContext(true, null);
+
+		ReactiveRedisTemplate<String, String> template = new ReactiveRedisTemplate(redisConnectionFactory, serializationContext);
+
+		ObjectRecord<String, String> record = template.opsForStream().read(String.class, StreamOffset.fromStart(streamKey)).blockFirst();
+		assertThat(record.getStream()).isEqualTo(streamKey);
+		assertThat(record.getValue()).isEqualTo(messagePayload);
+		template.delete(streamKey).block();
+	}
+
+	//TODO Find why the deserialization fail does not work
+	/*@Test
+	@RedisAvailable*/
+	public void explicitJsonSerializationContextTest() {
```

Comment:
@artembilan I need help here. I absolutely don't know why I don't have the result expected. I use the same serializationContext on both sides but the `read(...)` method returns a strange result. Thanks for your help